tests/test_datasetclient.py

```diff
@@ -0,0 +1,408 @@
+            segment.append(frame)
+
+        dataset_client.upload_segment(segment)
+        segment1 = FusionSegment(name="segment1", client=dataset_client)
+        assert len(segment1) == 10
+        assert segment1[0][LIDAR_DATA["name"]].path == "hello0.txt"
+        assert (
+            segment1[0][LIDAR_DATA["name"]].path
+            == segment[0][LIDAR_DATA["name"]].target_remote_path
+        )
+        assert not segment1[0][LIDAR_DATA["name"]].label
+        # todo: match the input and output label
+
+        gas_client.delete_dataset(dataset_name)
+
+    def test_get_and_upload_fusion_segment_object_with_label(self, accesskey, url, tmp_path):
+        gas_client = GAS(access_key=accesskey, url=url)
+        dataset_name = get_random_dataset_name()
+        dataset_client = gas_client.create_dataset(dataset_name, is_fusion=True)
+        dataset_client.create_draft("draft-1")
+        dataset_client.upload_catalog(Catalog.loads(CATALOG))
+
+        segment = FusionSegment("segment1")
+        segment.sensors.add(Sensor.loads(LIDAR_DATA))
+
+        path = tmp_path / "sub"
+        path.mkdir()
+        for i in range(10):
+            frame = Frame()
+            local_path = path / f"hello{i}.txt"
+            local_path.write_text("CONTENT")
+            data = Data(local_path=str(local_path))
+            data.label = Label.loads(LABEL)
+            frame[LIDAR_DATA["name"]] = data
+            segment.append(frame)
+
+        dataset_client.upload_segment(segment)
+        segment1 = FusionSegment(name="segment1", client=dataset_client)
+        assert len(segment1) == 10
+        assert segment1[0][LIDAR_DATA["name"]].path == "hello0.txt"
+        assert (
+            segment1[0][LIDAR_DATA["name"]].path
+            == segment[0][LIDAR_DATA["name"]].target_remote_path
+        )
+        assert segment1[0][LIDAR_DATA["name"]].label
+        # todo: match the input and output label
+
+        gas_client.delete_dataset(dataset_name)
+
+    def test_create_draft_data_inherit_top_commit(self, accesskey, url, tmp_path):
+        gas_client = GAS(access_key=accesskey, url=url)
+        dataset_name = get_random_dataset_name()
+        dataset_client = gas_client.create_dataset(dataset_name)
+        dataset_client.create_draft("draft-1")
+        segment = Segment("segment1")
+        path = tmp_path / "sub"
+        path.mkdir()
+        for i in range(10):
+            local_path = path / f"hello{i}.txt"
+            local_path.write_text("CONTENT")
+            data = Data(local_path=str(local_path))
+            segment.append(data)
+
+        dataset_client.upload_segment(segment)
+        dataset_client.commit("commit-1")
+
+        dataset_client.create_draft("draft-2")
+        segment1 = Segment(name="segment1", client=dataset_client)
+        assert len(segment1) == 10
+        assert segment1[0].get_url()
+        assert segment1[0].path == segment[0].target_remote_path
+
+        gas_client.delete_dataset(dataset_name)
+
```

Comment:
Union `test_create_draft_data_inherit_top_commit()` and `test_commit_draft_really_save_data()`~